docs/source/graphs.rst

```diff
@@ -97,3 +97,34 @@ special and others can write different schedulers better suited to other
 applications or architectures easily.  Systems that emit dask graphs (like
 Dask Array, Dask Bag, and so on) may leverage the appropriate scheduler for
 the application and hardware.
+
+
+Task Expectations
+-----------------
+
+When a task is submitted to Dask for execution, there are a number of assumptions
+that are made about that task. In general, tasks with side-effects that alter the
+state of a future in-place are not recommended. Values stored by Dask are mutable,
+and can be updated in-place. For example, consider a workflow involving a
```

Comment:
On second thought the sentence 

> Values stored by dask are mutable and can be updated in place

might be interpretted as 

> it is permissible to update data in place

raher than 

> nothing stops you from shooting yourself in the foot

Maybe instead of this sentence we say "Modifying data stored in Dask can have unintended consequences".

Reply:
Yes, I agree. I think there are probably cases where the behavior is non-deterministic or introduces race conditions, so maybe we should add that it's not officially a supported functionality, or explicitly mention the race condition possibility.